File: ProyectoFinal.Antares.Domain/Modelos/Pedido.cs
using ProyectoFinal.Antares.Domain.Enums;

namespace ProyectoFinal.Antares.Domain.Modelos;

public class Pedido : BaseModel
{
    public int IdUsuario { get; set; }
    
    public int? IdDelivery { get; set; }
    
    public virtual Usuario? Usuario { get; set; }

    public EstadoPedido EstadoPedido { get; set; }

    public List<PedidoProducto> ListaPedido { get; set; } 
    public virtual Usuario? Delivery { get; set; }
    
    public DateTime? HoraPedido { get; set; }
    
    public DateTime? HoraEntrega { get; set; }

    public decimal PrecioTotal { get; set; }

    public string Nota { get; set; } = null!;
}

File: FrontendWeb/ProyectoFinal_Antares_FrontEnd_Web/src/app/Models/pedido.ts
import { Usuario } from './usuario';
import { PedidoProducto } from './pedidoProducto';

export interface Pedido
{
    id: number;
    usuario: Usuario;
    delivery: Usuario;
    estadoPedido: EstadoPedido;
    listaPedido: PedidoProducto[];
    precioTotal: number;
    horaPedido: Date;
    horaEntrega: Date;
    nota: string;
}

export enum EstadoPedido
{
    Creado,
    Pagado,
    Preparando,
    EnCamino,
    Entregado,
    Finalizado
}
